xbmc/games/dialogs/osd/DialogInGameSaves.cpp

```diff
@@ -280,7 +280,7 @@ void CDialogInGameSaves::OnNewSave()
   CFileItemPtr item = std::make_shared<CFileItem>();
   CSavestateDatabase::GetSavestateItem(*savestate, savestatePath, *item);
 
-  m_savestateItems.AddFront(std::move(item), 0);
+  m_savestateItems.AddFront(item, 0);
```

Comment:
What about instead adding a move parameter? This change unfortunately looses context intended by the original programmer that shows the item is expected to be invalid at this point.

Reply:
I agree that the problem is with the `CFileItemList::AddFront(...)` method. I'd change it like this:
<details>
<summary>Patch suggestion</summary>

```diff
diff --git a/xbmc/FileItem.cpp b/xbmc/FileItem.cpp
index 2eb5eb072c..951d87ac0e 100644
--- a/xbmc/FileItem.cpp
+++ b/xbmc/FileItem.cpp
@@ -2250,21 +2250,22 @@ void CFileItemList::Add(CFileItem&& item)
   m_items.emplace_back(std::move(ptr));
 }
 
-void CFileItemList::AddFront(const CFileItemPtr &pItem, int itemPosition)
+void CFileItemList::AddFront(CFileItemPtr pItem, int itemPosition)
 {
   std::unique_lock<CCriticalSection> lock(m_lock);
 
-  if (itemPosition >= 0)
+  if (m_fastLookup)
   {
-    m_items.insert(m_items.begin()+itemPosition, pItem);
+    m_map.insert(MAPFILEITEMSPAIR(m_ignoreURLOptions ? CURL(pItem->GetPath()).GetWithoutOptions() : pItem->GetPath(), pItem));
   }
-  else
+
+  if (itemPosition >= 0)
   {
-    m_items.insert(m_items.begin()+(m_items.size()+itemPosition), pItem);
+    m_items.insert(m_items.begin()+itemPosition, std::move(pItem));
   }
-  if (m_fastLookup)
+  else
   {
-    m_map.insert(MAPFILEITEMSPAIR(m_ignoreURLOptions ? CURL(pItem->GetPath()).GetWithoutOptions() : pItem->GetPath(), pItem));
+    m_items.insert(m_items.begin()+(m_items.size()+itemPosition), std::move(pItem));
   }
 }
 
diff --git a/xbmc/FileItem.h b/xbmc/FileItem.h
index e0adc28e4d..09bdae4226 100644
--- a/xbmc/FileItem.h
+++ b/xbmc/FileItem.h
@@ -727,7 +727,7 @@ public:
   void ClearItems();
   void Add(CFileItemPtr item);
   void Add(CFileItem&& item);
-  void AddFront(const CFileItemPtr &pItem, int itemPosition);
+  void AddFront(CFileItemPtr pItem, int itemPosition);
   void Remove(CFileItem* pItem);
   void Remove(int iItem);
   CFileItemPtr Get(int iItem) const;
```
</details>